python/lsst/ap/association/trailedSourceFilter.py

```diff
@@ -79,32 +84,42 @@ def run(self, dia_sources, exposure_time):
             exceed max_trail_length/second*exposure_time.
             (`pandas.DataFrame`)
         """
-        trail_mask = self._check_dia_source_trail(dia_sources, exposure_time)
+
+        flag_map = os.path.join(utils.getPackageDir("ap_association"), "data/association-flag-map.yaml")
+        unpacker = UnpackApdbFlags(flag_map, "DiaSource")
+        flags = unpacker.unpack(dia_sources["flags"], "flags")
+
+        trail_mask = self._check_dia_source_trail(dia_sources, exposure_time, flags)
 
         return pipeBase.Struct(
             diaSources=dia_sources[~trail_mask].reset_index(drop=True),
             trailedDiaSources=dia_sources[trail_mask].reset_index(drop=True))
 
-    def _check_dia_source_trail(self, dia_sources, exposure_time):
+    def _check_dia_source_trail(self, dia_sources, exposure_time, flags):
         """Find DiaSources that have long trails.
 
         Return a mask of sources with lengths greater than
-        ``config.max_trail_length``  multiplied by the exposure time.
+        ``config.max_trail_length``  multiplied by the exposure time and
+        have ext_trailedSources_Naive_flag_edge set.
```

Comment:
Why is this an "and" and not an "or" situation?

Reply:
Your right, its an or, possibly an and/or situation. Updated it to or.